### PR TITLE
Add a recipe for setting up a virtualenv

### PR DESCRIPTION
**Pull request by mwichmann**
Signed-off-by: Mats Wichmann <mats@linux.com>

### CODE REVIEW

**Comment by mwichmann** [comment]
Is it important to include an equivalent log of doing on Windows, btw?

File: docs/recipes/virtualenv.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+isolated execution environment - you can install whatever you need
+there, and change versions as needed, without affecting anything
+else on your system that uses Python.  Here is an example session
+for setting one up:
+
+.. code:: shell
+
+   $ cd Work
+   $ python -m venv myvenv
+   $ cd myvenv
```
Comment (bdbaddog) [inline docs/recipes/virtualenv.rst:15]: You shouldn't need to cd into myenv to source bin/activate

For windows you use scripts/activate.bat or activate.ps1(for powershell)

File: docs/recipes/virtualenv.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   $ source bin/activate
+   (myvenv) $ python -m pip list --outdated
```
Comment (bdbaddog) [inline docs/recipes/virtualenv.rst:17]: Only time I use `python -m pip` is on windows to upgrade pip itself

I'd just use pip here.

File: docs/recipes/virtualenv.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   Package    Version Latest Type
+   ---------- ------- ------ -----
+   pip        19.3.1  21.0.1 wheel
+   setuptools 41.6.0  53.0.0 wheel
+   WARNING: You are using pip version 19.3.1; however, version 21.0.1 is available.
+   You should consider upgrading via the 'pip install --upgrade pip' command.
+   (myvenv) $ python -m pip install --upgrade pip setuptools scons
```
Comment (bdbaddog) [inline docs/recipes/virtualenv.rst:24]: But not here, and maybe include a comment about why.. though `pip install -U pip` works on non-windows, it fails on windows.

**Comment by bdbaddog** [comment]
I'd replace all the `python -m pip`'s with `pip` except for upgrading pip itself.

File: docs/recipes/virtualenv.rst (hunk 2 above)
Comment (mwichmann) [reply docs/recipes/virtualenv.rst:17]: I can change that; if the virtualenv is active, pip indeed will be "the correct one", I just usually use this form when showing people stuff in order to cultivate good habits.  People get itno a *lot* of trouble just running pip, and then "why won't Python find my module".

File: docs/recipes/virtualenv.rst (hunk 3 above)
Comment (mwichmann) [reply docs/recipes/virtualenv.rst:24]: Took a moment to parse this.  Anyway, I have the equivalent Windows log from the same time, just wasn't sure if it needed to be added to the recipe.
```
C:\Users\mats>cd Work

C:\Users\mats\Work>py -m venv myvenv

C:\Users\mats\Work>cd myvenv

C:\Users\mats\Work\myvenv>.\Scripts\activate

(myvenv) C:\Users\mats\Work\myvenv>python -m pip list --outdated
Package    Version Latest Type
---------- ------- ------ -----
pip        20.2.3  21.0.1 wheel
setuptools 49.2.1  53.0.0 wheel
WARNING: You are using pip version 20.2.3; however, version 21.0.1 is available.You should consider upgrading via the 'C:\Users\mats\Work\myvenv\Scripts\python.exe -m pip install --upgrade pip' command.

(myvenv) C:\Users\mats\Work\myvenv>python -m pip install --upgrade pip setuptools scons
Collecting pip
  Using cached pip-21.0.1-py3-none-any.whl (1.5 MB)
Collecting setuptools
  Using cached setuptools-53.0.0-py3-none-any.whl (784 kB)
Collecting scons
  Using cached SCons-4.1.0.post1-py3-none-any.whl (4.1 MB)
Installing collected packages: pip, setuptools, scons
  Attempting uninstall: pip
    Found existing installation: pip 20.2.3
    Uninstalling pip-20.2.3:
      Successfully uninstalled pip-20.2.3
  Attempting uninstall: setuptools
    Found existing installation: setuptools 49.2.1
    Uninstalling setuptools-49.2.1:
      Successfully uninstalled setuptools-49.2.1
Successfully installed pip-21.0.1 scons-4.1.0.post1 setuptools-53.0.0

(myvenv) C:\Users\mats\Work\myvenv>scons --version
SCons by Steven Knight et al.:
        SCons: v4.1.0.post1.dc58c175da659d6c0bb3e049ba56fb42e77546cd, 2021-01-20 04:32:28, by bdbaddog on ProDog2020
        SCons path: ['c:\\users\\mats\\work\\myvenv\\lib\\site-packages\\SCons']
Copyright (c) 2001 - 2021 The SCons Foundation

(myvenv) C:\Users\mats\Work\myvenv>
```

File: docs/recipes/virtualenv.rst [hunk 4]
```diff
@@ -6,22 +6,21 @@ that uses SCons to build.  A virtualenv is a way to create an
...
 else on your system that uses Python.  Here is an example session
-for setting one up:
+for setting one up on a Linux host:
 
 .. code:: shell
 
    $ cd Work
    $ python -m venv myvenv
-   $ cd myvenv
    $ source bin/activate
```
Comment (bdbaddog) [inline docs/recipes/virtualenv.rst:15]: `source myvenv/bin/activate`

Comment (bdbaddog) [reply docs/recipes/virtualenv.rst:15]: and actually `. myvenv/bin/activate` should also work. (that's what I typically do, less typing.. )

Comment (mwichmann) [reply docs/recipes/virtualenv.rst:15]: yeah, hand-editing logs never works out right

**Comment by bdbaddog** [comment]
A thought.. perhaps a TL:DR section at the top with just
```
python -m venv myenv
. bin/activate
pip install SCons
scons --version
```

Then the current text as an expanded info below?

File: docs/recipes/virtualenv.rst (hunk 4 above)
Comment (mwichmann) [reply docs/recipes/virtualenv.rst:15]: I can trim these further if you want - just trying to figure out what still needs doing here to complete this.

Comment (bdbaddog) [reply docs/recipes/virtualenv.rst:15]: Yup. If you can fix that, then we're good to go.

**Comment by mwichmann** [comment]
I see I forgot to make a new branch for this PR, is it still going to be okay, or should I start over?

**Pull request merged**